scripts/test_hardware/test_esp32_api.py

```
@@ -0,0 +1,331 @@
+REQUEST_TIMEOUT = 5  # seconds
+
+
+def _get(url: str, timeout: float = REQUEST_TIMEOUT) -> tuple[int, str]:
+    """HTTP GET, returns (status_code, body)."""
+    req = urllib.request.Request(url)
+    try:
+        with urllib.request.urlopen(req, timeout=timeout) as resp:
+            return resp.status, resp.read().decode("utf-8", errors="replace")
+    except urllib.error.HTTPError as e:
+        return e.code, e.read().decode("utf-8", errors="replace")
+    except Exception as e:
```

Ruff (0.15.2)

[error] Local variable `e` is assigned to but never used

Remove assignment to unused variable `e`

(F841)